### PR TITLE
Fix assembly duplicities in metadata browser

### PR DESCRIPTION
**Pull request by acizmarik**
This PR fixes an issue with assembly duplicities in metadata browser. We are filtering assemblies based on a 2-item key `(assemblyName, assemblyVersion)`. This filters situations such as loading `System.Data.dll` and `System.Data.ni.dll` and having almost identical entries in the metadata browser.

Please merge first #13

### CODE REVIEW

File: src/Meditation.MetadataLoaderService/Services/MetadataLoader.cs [hunk 1]
```diff
@@ -4,36 +4,60 @@
...
         }
 
-        public AssemblyMetadataEntry LoadMetadataFromAssembly(string path)
+        public IEnumerable<AssemblyMetadataEntry> LoadMetadataFromProcess(IEnumerable<string> modulePaths)
+        {
+            var assembliesLookup = new Dictionary<(UTF8String Name, Version Version), AssemblyDef>();
+            foreach (var modulePath in modulePaths.OrderBy(Path.GetFileName))
+            {
+                var (metadata, assemblyDef) = LoadMetadataFromAssemblyImpl(modulePath);
+                if (assembliesLookup.ContainsKey((assemblyDef.Name, assemblyDef.Version)))
```
Comment (LunarWhisper) [inline src/Meditation.MetadataLoaderService/Services/MetadataLoader.cs:31]: [Question] What if we'll get NGEN image first? Is it ok? Do you need only name/version?

File: src/Meditation.MetadataLoaderService/Services/MetadataLoader.cs [hunk 2]
```diff
@@ -4,36 +4,60 @@
...
+                {
+                    // This means that the assembly is already processed
+                    // It happens usually in connection with NGEN (see: https://learn.microsoft.com/en-us/dotnet/framework/tools/ngen-exe-native-image-generator).
+                    // For example, consider case when we are processing modules: System.Data.dll and subsequently try to process System.Data.ni.dll
+
+                    // FIXME: log reason for skipping assembly processing
```
Comment (LunarWhisper) [inline src/Meditation.MetadataLoaderService/Services/MetadataLoader.cs:37]: [ToFix] Create an Issue and add link to it, please.

**Review by LunarWhisper — CHANGES_REQUESTED** [review]
One question and a little request.

**Review by aleksylum — APPROVED** [review]

File: src/Meditation.MetadataLoaderService/Services/MetadataLoader.cs (hunk 1 above)
Comment (acizmarik) [reply src/Meditation.MetadataLoaderService/Services/MetadataLoader.cs:31]: > What if we'll get NGEN image first? Is it ok?

It is OK. Getting NGEN image first is normal (or in fact getting only native image is normal). They have identical metadata entries so we do not really care which is loaded. Native images are supposed to be just performance optimization. Runtime seems to prefer native images when they are available: https://github.com/dotnet/runtime/blob/5f4ee519c7ce0bfe1e914ee7a617e17b5c509cf5/src/coreclr/binder/utils.cpp#L185

> Do you need only name/version?

According to documentation if we want to implement correct behavior, we should also consider culture and public key (if it is available). See this: https://learn.microsoft.com/en-us/dotnet/framework/deployment/how-the-runtime-locates-assemblies

**Review by LunarWhisper — APPROVED** [review]
LGTM

**Pull request merged**